### ISSUE TITLE
The 2017 data period  with Turkish support

### ISSUE THREAD

**Issue by queeup** [Spidersouris/termic#4]
Hi,

Thank you for this wonderful project. Is it possible to add Turkish language to 2017 data? I really need Windows string search. Their 2020+ data is very limited. I miss their old terminology search website.

**Comment by spidersouris** [comment]
Hi @queeup! Thank you for your kind words.

I'm sorry to hear that data is limited for the 2020+ period. Right now I'm not really pleased with how the 2017 data is implemented as it's taking a very long time to search in the DB and I'm not even sure the request is able to proceed successfully most of the time. Unfortunately I haven't had a lot of time to work on termic these last few months and I couldn't improve this side of things as I had planned to.

I will try to add 2017 data for Turkish in the next couple of weeks but I'm not sure this is going to work out; as in if search will be efficient enough and if it will even work properly.

**Comment by queeup** [comment]
Thank you for adding it. Do you know anywhere that I can download 2017 data with windows localization included? I don't think official microsoft language portal search included windows strings.

**Comment by spidersouris** [comment]
Sure! The README includes a Dropbox link with all the data used for termic. You should be able to find the 2017 data in the 2017 folder as tk_ui_strings_glossary_january_2017_x86_x64_9828500.zip. Can you confirm you can access it and it contains the data that you're searching for?

**Comment by queeup** [comment]
I have it but unfortunately there is no Turkish data. tk_* is not Turkish. It must be tr_*.... You don't have Turkish data in `termic_data/TranslationAndUIStrings/2017` dropbox folder.

**Comment by queeup** [comment]
I think Turkish data is in 2018. I found the file name but no download available online.`tr_ui_strings_glossary_july_2018_x86_x64_12866510.zip`

**Comment by spidersouris** [comment]
You're right. Actually I recall now that when searching 2017 files I couldn't find data for some languages, Turkish included. I wrote about it on my blog back in July:

> Currently, the 2017 data period option is only available for VSCode-supported languages (Czech, English, French, German, Italian, Japanese, Korean, Polish, Portuguese (Brazil), Russian, Simplified Chinese, Spanish (Spain) and Traditional Chinese). Turkish is not available because that language’s files are missing from the 2017 data repository.
- https://blog.edoyen.com/post/new-data-for-termic-and-plans-for-the-future/

I'm not sure to what extent I may be able to find the missing data for Turkish but I can try to take a look at it. Perhaps the 2018 data will be available; thanks for the clue! I'll keep you updated.
